ambry-store/src/test/java/com.github.ambry.store/LogTest.java

```diff
@@ -473,7 +476,10 @@ private void setupAndDoComprehensiveTest(long logCapacity, long segmentCapacity,
       segmentNames.add(name);
     } else {
       for (int i = 0; i < numToCreate; i++) {
-        long pos = Utils.getRandomLong(TestUtils.RANDOM, 1000);
+        long pos;
+        do {
+          pos = Utils.getRandomLong(TestUtils.RANDOM, 1000);
```

Comment:
This looks like it'll go into an infinite loop if numToCreate goes over a 1000; I'm guessing that's an unfeasible scenario?  Maybe add an illegalArgumentException that triggers when numToCreate goes over 1000 (or make 1000 a final static variable itself)

Reply:
Could also replace this with an array of 0-999 that you randomly pull from then shuffle used numbers toward the back... probably overkill for this test though

Reply:
It is an unfeasible scenario - going to leave this as is for now but I do think thats a good idea for something thats production facing.